rcl_yaml_param_parser/test/test_parse_yaml.cpp

```diff
@@ -43,6 +43,26 @@ TEST(test_file_parser, correct_syntax) {
   allocator.deallocate(path, allocator.state);
 }
 
+TEST(test_file_parser, string_array_with_quoted_number) {
+  rcutils_reset_error();
+  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024));
+  char * test_path = rcutils_join_path(cur_dir, "test", allocator);
+  char * path = rcutils_join_path(test_path, "string_array_with_quoted_number.yaml", allocator);
+  fprintf(stderr, "cur_path: %s\n", path);
+  EXPECT_TRUE(rcutils_exists(path));
+  rcl_params_t * params_hdl = rcl_yaml_node_struct_init(allocator);
+  EXPECT_TRUE(params_hdl);
+  if (params_hdl) {
+    bool res = rcl_parse_yaml_file(path, params_hdl);
+    fprintf(stderr, "%s\n", rcutils_get_error_string().str);
+    EXPECT_TRUE(res);
+    rcl_yaml_node_struct_print(params_hdl);
+    rcl_yaml_node_struct_fini(params_hdl);
+  }
+  free(test_path);
+  free(path);
```

Comment:
Now that we're using an allocator, these two lines should be changed to use `allocator.deallocate()`.